frontend/src/App.tsx

```diff
@@ -841,6 +841,23 @@ export class App extends PureComponent<Props, State> {
       )
       this.widgetMgr.removeInactive(activeWidgetIds)
 
+      // Tell the ConnectionManager to increment the message cache run
+      // count. This will result in expired ForwardMsgs being removed from
+      // the cache.
+      if (this.connectionManager !== null) {
+        this.connectionManager.incrementMessageCacheRunCount(
+          SessionInfo.current.maxCachedMessageAge
+        )
```

Comment:
This is copied from the FINISHED_SUCCESSFULLY branch just above, to keep this branch as "the FINISHED branch minus the widget clearing". Because the lines are identical, git displays this as adding this chunk into the above branch and using the already existing copy in the new branch.